modules/solid_mechanics/src/materials/ReturnMappingModel.C

```diff
@@ -186,3 +190,16 @@ ReturnMappingModel::computeStress(const Elem & /*current_elem*/,
 
   computeStressFinalize(qp, inelastic_strain_increment);
 }
+
+Real
+ReturnMappingModel::computeTimeStepLimit(unsigned qp)
+{
+  Real tmp_max_incr;
+  Real scalar_inelastic_strain_incr;
+
+  scalar_inelastic_strain_incr =
+      _effective_inelastic_strain[qp] - _effective_inelastic_strain_old[qp];
+  tmp_max_incr = _max_inelastic_incr * 1e-10;
+  _max_inc = std::max(scalar_inelastic_strain_incr, tmp_max_incr);
```

Comment:
I don't see any reason for `_max_inc` to be a member variable because this is the only place it is used. Just make it local to this method.  Also, I find this code hard to follow.  There's a magic factor of 1e-10 coming out of the blue, and _max_inc seems to be mixing the current strain increment and the allowable increment somehow. Is this just about avoiding a divide by zero?  If so, it would be better to just check if the denominator is close to zero (using `MooseUtils::absoluteFuzzyEqual(_max_inc, 0.0`), and return `std::numeric_limits<Real>::max()` if it is.